src/ApiService/ApiService/Log.cs

```diff
@@ -1,14 +1,57 @@
 ﻿using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Text;
 using Microsoft.ApplicationInsights;
 using Microsoft.ApplicationInsights.DataContracts;
 using Microsoft.ApplicationInsights.Extensibility;
 
 namespace Microsoft.OneFuzz.Service;
 
+//See: https://learn.microsoft.com/en-us/dotnet/csharp/whats-new/tutorials/interpolated-string-handler
+//by ref struct works, but Moq does not support by-ref struct value so all the tests break... https://github.com/moq/moq4/issues/829
+[InterpolatedStringHandler]
+public struct LogStringHandler {
+
+    private StringBuilder _builder;
+    private Dictionary<string, string>? _tags;
+
+    public LogStringHandler(int literalLength, int formattedCount) {
+        _builder = new StringBuilder(literalLength);
+        _tags = null;
+    }
+
+    public void AppendLiteral(string message) {
+        _ = _builder.Append(message);
```

Comment:
Instead of having to discard, you can add StringBuilder to the CSharpExtensions config file